File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    let primes = 0;
    
    for(let i = 2; i <= n; i++){
        
        if(!checkPN(i)) {
            primes++
        }
    }
    
    console.log(primes)
    return primes;
}

const checkPN = (num) => {
    let isDivided = false;
    
    for(let p = 2; p <= Math.floor(Math.sqrt(num)); p++){
        if(num % p === 0) {
            isDivided = true;    
            break;
        }
    }
    
    return isDivided
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
import math

def isPrime (n):
    if n == 2:
        return True
    for i in range(2, math.isqrt(n) + 1):
        if n % i == 0:
            return False
    return True

def solution(n):
    
    return sum(1 for i in range(2, n + 1) if isPrime(i))
        

